### PR TITLE
Avoid writing tasks inside functions by raising errors

### PR DESCRIPTION
**Pull request by kumare3**

### CODE REVIEW

**Comment by wild-endeavor** [comment]
I'm gonna look at this later, this won't affect the beta release right?

File: flytekit/annotated/python_function_task.py [hunk 1]
```diff
@@ -122,6 +122,38 @@ def get_container(self, settings: RegistrationSettings) -> _task_model.Container
         )
 
 
+def isnested(func) -> bool:
```
Comment (katrogan) [inline flytekit/annotated/python_function_task.py:125]: extreme nit but can we use snake case for these method names for readability?

File: tests/flytekit/unit/annotated/test_type_hints.py [hunk 2]
```diff
@@ -984,3 +984,19 @@ def my_subwf():
...
+    @task
+    def foo():
+        pass
+
+    import inspect
+
+    def is_nested(func) -> bool:
+        return func.__code__.co_flags & inspect.CO_NESTED != 0
+
+    print("\n")
```
Comment (katrogan) [inline tests/flytekit/unit/annotated/test_type_hints.py:999]: add assertions?

**Review by katrogan — APPROVED** [review]

File: flytekit/annotated/python_function_task.py (hunk 1 above)
Comment (kumare3) [reply flytekit/annotated/python_function_task.py:125]: i just followed the style in inspect. I was going to use snake case, what do you think
they use. issubclass, isclass, isobject etc

File: tests/flytekit/unit/annotated/test_type_hints.py (hunk 2 above)
Comment (kumare3) [reply tests/flytekit/unit/annotated/test_type_hints.py:999]: aah delete

File: flytekit/annotated/python_function_task.py (hunk 1 above)
Comment (katrogan) [reply flytekit/annotated/python_function_task.py:125]: TIL. ignore me then

Comment (kumare3) [reply flytekit/annotated/python_function_task.py:125]: resolving

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**